### ISSUE TITLE
POST method returns RedirectResponse when validation throws error

### ISSUE THREAD

**Issue by letschke** [mcamara/laravel-localization#367]
Hello,

I have a simple newsletter signup form where you can insert an email address and will receive a newsletter. If you don't enter an email address or if the email address is not valid the previous form will be shown again with an error description.

However after integrating mcamara/laravel-localization I get a Laravel error when validation is executed.
HttpResponseException -> RedirectResponse

> statusCode: 302

My SubscriberRequest.php has validation in it so that an email address is mandatory:

``` php
public function rules()
    {
        return [
            'email' => 'required|email',
        ];
    }

public function messages()
    {
        return [
            'email.required'  => 'Please enter your email address.',
            'email.email'  => 'Your email address is not valid.',
        ];
    }
```

Here is my route configuration:

``` php
Route::group(['prefix' => LaravelLocalization::setLocale(),'middleware' => ['localeSessionRedirect', 'localizationRedirect', 'web']], function () {
    Route::group(['namespace' => 'Frontend'], function() {
        Route::get('newsletter', ['as' => 'newsletter', 'uses' => 'SubscriberController@create']);
        Route::post('newsletter', ['as' => 'newsletter_store', 'uses' => 'SubscriberController@store']);
    });
});
```

This the SubscriberController:

``` php
public function store(SubscriberRequest $request)
    {
    Subscriber::create($request->all());
    return redirect('newsletter')
        ->with('success_message', 'Subscription successful.');
    }
```

And here is my form:

``` php
{!! Form::open(array('url' => url(LaravelLocalization::getLocalizedURL(LaravelLocalization::getCurrentLocale(), '/newsletter')), 'class' => 'send-form', 'novalidate' => 'novalidate')) !!}

    <div class="form-group {{ $errors->has('email') ? 'has-error' : ''}}">
        {!! Form::label('email', 'E-Mail*: ', ['class' => '']) !!}
        {!! Form::text('email', null, ['required', 'class' => 'form-control']) !!}
        {!! $errors->first('email', '<p class="help-block">:message</p>') !!}
    </div>

    <div class="form-group">
        {!! Form::submit('Order newsletter', ['class' => 'btn btn-primary btn-wd']) !!}
    </div>
{!! Form::close() !!}
```

I suppose it should be fairly easy but I just can't get it to work. I have read about redirecting to 307 instead of 302 but can't that to work either.

I'm using Laravel 5.2.

Any help would be appreciated. Thank you!

**Comment by ctf0** [comment]
i believe ur web middleware should be **before** the other middlewares , because that's the one responsible for applying the session/auth/etc...

**Comment by letschke** [comment]
Hello ctf0,

thanks for the hint but unfortunately changing order of the middlewares didn't fix the error. Still the same. Do you have another idea?

Here my current routes info:
```php
Route::group(['prefix' => LaravelLocalization::setLocale(),'middleware' => ['web', 'localeSessionRedirect', 'localizationRedirect']], function () {
    Route::group(['namespace' => 'Frontend'], function() {
        Route::get('newsletter', ['as' => 'newsletter', 'uses' => 'SubscriberController@create']);
        Route::post('newsletter', ['as' => 'newsletter_store', 'uses' => 'SubscriberController@store']);
    });
});
```

**Comment by ctf0** [comment]
oh my bad, even though it still wont work because `LaravelLocalization::setLocale()` use session, so instead try 

```php
Route::group(['middleware' => 'web'], function () {
    
    // all ur routes should be inside here

    Route::group(['prefix' => LaravelLocalization::setLocale(),'middleware' => ['localeSessionRedirect', 'localizationRedirect']], function () {

        // any localized routes should be here

        Route::group(['namespace' => 'Frontend'], function() {
            Route::get('newsletter', ['as' => 'newsletter', 'uses' => 'SubscriberController@create']);
            Route::post('newsletter', ['as' => 'newsletter_store', 'uses' => 'SubscriberController@store']);
        });
    });
});
```

in v5.3 the web issue was solved through applying it to the whole routes.php file

**Comment by letschke** [comment]
Hello ctf0,

had the same issue on Laravel 5.3.

I have now added handling of validation errors in the render function of the app/Exceptions/Handler.php in order to return to the view directly and show possible validation errors to the user.

`
if ($exception instanceof ValidationException) {
    return parent::render($request, $exception);
}
`

This has resolved the issue in Laravel 5.2 as well as 5.3

Thanks for your help!

**Comment by ctf0** [comment]
the validation exception is handled by default, but if u r using https://github.com/GrahamCampbell/Laravel-Exceptions , then u do need the extra line as its not supported by default
